### ISSUE TITLE
Bug: `cabal repl file` fails for executable component

### ISSUE THREAD

**Issue by fendor** [haskell/cabal#6622]
**To Reproduce**
Set up an example project:
```
$  cabal init -p example --libandexe --tests --main-is=Main.hs --source-dir=src --application-dir=app --test-dir=test
```
Load library component:
```
$ cabal repl src/MyLib.hs
# Succeeds to load
$ cabal repl src/Lib.hs
# Fails to load as expected 
$ cabal repl app/Main.hs
# Fails to load although
cabal.exe: Unknown target './app/Main.hs'.
The package example has no file target 'app/Main.hs'.
$ cabal repl test/MyTestLib.hs
cabal.exe: Unknown target './test/MyLibTest.hs'.
The package example has no file target 'test/MyLibTest.hs'.
```
**Expected behavior**

The file targets `app/Main.hs` and `test/MyTestLib.hs` should successfully load.

**System information**
 - `cabal`: `3.0.0.0`, but also HEAD.

**Benefit**

Directly benefits [hie-bios](https://github.com/mpickering/hie-bios) which uses `cabal repl <filename>` to find compilation flags.
cc @mpickering

**Comment by fendor** [comment]
I have the bug fix almost ready. It is a bug in the target resolvement.

**Comment by mouse07410** [comment]
Naive question: when can one hope to see this PR merged, and what Cabal release will it end up in?

**Comment by fendor** [comment]
Hopefully next week, and Ill assume cabal 3.4?

**Comment by mouse07410** [comment]
Hackage now serves Cabal 3.2.0.0.

Do you have an idea of the release rhythm? I.e., how long works it normally take to release 3.4? Several months? Weeks? Years? 

And fixes are rarely/never backported?

**Comment by phadej** [comment]
@mouse07410 there is no rhythm. So far Cabal releases were tied to GHC releases, but I don't know what are the plans for GHC-8.12, they are not on GHC status page: https://gitlab.haskell.org/ghc/ghc/-/wikis/status

**Comment by mouse07410** [comment]
@phadej  thanks for explaining.

You probably know if "maintenance releases" happen? I.e., is there a chance that this PR (and possibly other fixes) would be released before Cabal 3.4 is ready?

**Comment by fendor** [comment]
Fixed by #6826
